user_tools/src/spark_rapids_tools/tools/unsupported_ops_stage_duration.py

```diff
@@ -42,7 +42,9 @@ def prepare_apps_with_unsupported_stages(self, all_apps: pd.DataFrame,
         result_df[result_col_name] = result_df[result_col_name].fillna(0)
         # Update the percentage column
         perc_result_col_name = self.props.get('percentResultColumnName')
-        result_df[perc_result_col_name] = result_df[result_col_name] * 100.0 / result_df['App Duration']
+        # Calculate the percentage of total application duration, clip the value to 100.0
+        result_df[perc_result_col_name] = ((result_df[result_col_name] * 100.0 / result_df['App Duration'])
```

Comment:
do we have a follow on to really fix this?  Really I would have expected this to be in the java side of things.

Reply:
Thanks @tgravescs. I agree we are adding wall clock time and then comparing with app duration might not make sense totally.  The goal of this estimate is to filter out applications having stages with large amount of unsupported operators. 

@kuhushukla can provide more insights on the usage.

Reply:
Clipping does not seem the answer to it. Patching it means that the TCL could list incorrect results.

Reply:
Agree with Ahmed and Tom that we should fix it thru fixing the inherent issue with diration calculations.

Reply:
We would have to come up with a different logic for unsupported operator stage duration in that case. 

Can we use `Unsupported Stages Duration / All Stages Duration` instead of `App Duration`?

Reply:
I was about to suggest the same thing.
`Unsupported Stages Duration / All Stages Duration` will remove the WallClockTime from the equation.
However, this means that the threshold needs to be re-evaluated (25%).
IIRC, @partho you have evaluated TCL view on some eventlogs. Perhaps, we can re-evaluate them after changing the denominator

Reply:
I tested the above logic on a set of event logs and the updated numbers make more sense now. I am using the following approach:

1. Add column `Total Stage Duration` in `rapids_4_spark_qualification_output_unsupportedOperators.csv`.
2. Group apps by app id and calculate percentage as `unsupported operators stage duration / total stage duration`
3. Merge this dataframe with all_apps information
4. Now there are two things to consider:
   1. There may be apps that no unsupported operators, we will not have entries for them -> Unsupported Op Percentage should be 0% for these apps.
   2. There may be apps that have unsupported operators but have no impact (ie. `IgnoreNoPerf` case) -> Unsupported Op Percentage should be 0% for these apps.